tests/cbmc/stubs/madvise.c

```diff
@@ -14,8 +14,16 @@
  */
 
 #include <cbmc_proof/nondet.h>
+#include <errno.h>
+
+#include "error/s2n_errno.h"
 
 int madvise(void *addr, size_t length, int advice)
 {
-    return nondet_int();
+    assert(S2N_MEM_IS_WRITABLE(addr,length));
+    if(nondet_bool()) {
+        return 0;
+    }
+    errno = nondet_int();
+    return -1;
```

Comment:
Rather than returning the error number, you are setting the global errno variable, is this correct?

Reply:
Yes, it is. I'm implementing as specified in the [documentation](https://linux.die.net/man/2/madvise).
```c
On success madvise() returns zero. On error, it returns -1 and errno is set appropriately.
```